File: PrefixSum/pivot_index.py
def pivot_index(nums: list[int]) -> int:
    total = sum(nums)
    left = 0

    for i, num in enumerate(nums):
        right = total - left - num

        if left == right:
            return i
        
        left += num

    return -1


print(pivot_index([1,7,3,6,5,6]))

File: Arrays&Strings/PrefixSum/PivotIndex.js
/**
 * @param {number[]} nums 
 */
function PivotIndex(nums) {
    const total = nums.reduce((prev, acc) => prev + acc, 0);
    let left = 0;

    for (let i = 0; i < nums.length; i++) {
        const right = total - left - nums[i];

        if (left === right) return i;

        left += nums[i];
    }

    return -1;
}

console.log(PivotIndex([1,7,3,6,5,6]))